pallets/parachain-staking/src/lib.rs

```diff
@@ -1448,6 +1480,41 @@ pub mod pallet {
 			});
 			Ok(().into())
 		}
+
+		/// Batch migrate locks to freezes for a list of accounts.
+		///
+		/// This function allows migrating multiple accounts from the old lock-based
+		/// staking to the new freeze-based staking in a single transaction.
+		///
+		/// Parameters:
+		/// - `accounts`: List of account IDs to migrate
+		/// - `is_collator`: Whether the accounts are collators (true) or delegators (false)
+		///
+		/// The maximum number of accounts that can be migrated in one batch is 100.
+		#[pallet::call_index(33)]
+		#[pallet::weight(
+			T::WeightInfo::migrate_locks_to_freezes_batch(accounts.len() as u32)
+		)]
+		pub fn migrate_locks_to_freezes_batch(
+			origin: OriginFor<T>,
+			accounts: Vec<T::AccountId>,
+			is_collator: bool,
```

Comment:
This boolean should be per account instead of global, I don't see any reason to force a batch to contains only one kind of account

Reply:
Amended https://github.com/moonbeam-foundation/moonbeam/pull/3306/commits/3cd5365a8a80a3ec9b0e3c2601fe4af49c3b0775